doc/mgmt/Management Framework.md

```diff
@@ -2378,54 +2469,68 @@ mechanism for the app specific configuration.
+24. App modules perform writes of the translated data to the DB access layer.
+25. DB access layer validates the writes using CVL and then caches them.
+26. Status being returned from DB access layer.
+27. Status being returned from App module.
+28. Translib infra invokes the commit transaction on the DB access layer.
+29. DB access layer first invokes MULTI request on the Redis DB indicating there are multiple writes coming in, so commit everything together. All writes succeed or nothing succeeds.
+30. Status returned from Redis.
+31. Pipeline of all the cached writes are executed from the DB access layer.
+32. Status retuned from Redis.
+33. EXEC call is made to the Redis DB. Here if the call fails, it indicates that one of the keys that we watched has changed and none of the writes will go into the Redis DB.
+34. Status returned from Redis DB.
+35. Status returned from DB access layer.
+36. Write lock acquired in Step 6 is released.
+37. Status returned from the Translib infra.
+38. REST Status returned from the Request handler.
+39. REST response is sent by the REST gateway to the REST client.
 
 ### 4.2 REST GET flow
 
-![REST GET flow](images/read.jpg)
+![REST GET flow](images/read_mdb.jpg)
 
 1. REST GET request from the REST client is sent to the REST Gateway.
 2. REST Gateway invokes a common request handler.
 3. Authentication of the incoming request is performed.
 4. Request handler calls the Translib exposed GET API with the uri of the request.
 5. Translib infra gets the App module corresponding to the incoming uri.
-6. Translib infra calls the initialize function of the App module with the YGOT structures and path. App module caches them.
-7. Status retuned from App module.
-8. App module queries Transformer to translate the path to the Redis keys that need to be queried.
-9. Status returned from App module.
-10. Translib infra calls the processGet function on the App module
-11. App modules calls read APIs exposed by the DB access layer to read data from the Redis DB.
-12. Data is read from the Redis DB is returned to the App module
-13. App module fills the YGOT structure with the data from the Redis DB and validated the filled YGOT structure for the syntax.
-14. App module converts the YGOT structures to JSON format.
-15. IETF JSON payload is returned to the Translib infra.
-16. IETF JSON payload is returned to the request handler.
-17. Response is returned to REST gateway.
-18. REST response is returned to the REST client from the REST gateway.
+6. Translib infra calls the getNamespace function of the App module with the uri path, to get all DB namespace associated with the uri path
+7. App module queries Transformer to get all namespace associated with the uri path
+8. Transformer shall use Yang specific "Transformer Spec" to get all namespace associated with uri path
+9. Transformer return namespace associated with the uri path
+10. App module return namespace retrived by the transformer layer
```

Comment:
If name-space can be annotated to any yang-node/subpath in  a given yang-module and the associated namespace(s) is retrieved based on the request URL target ,at the very beginning in translib layer before the processing of the query is done in transformer layer what happens in the following case?

Yang-module openconfig-x has following structure with list node-C annotated with name-space and the REST GET request URL is at list node-A(openconfig-x/container-As/list-A) that has no namespace associated with it and is parent of list node-C :
openconfig-x 
    container-As
       list-A 
         ...
         ...
         container-Cs
             list-C    **name-space-annotation**
            
 As per the points mentioned in this section only the default namespace will be returned since the URL is at list-A. Transformer infra processing/traversal of the child yang hierarchy happens much later in processGet(point 16.)  and the node annotated with name-space will be known/reached much later. How will this be handled?   The name-space seems to rely only on URL/reuest target.

Reply:
> If name-space can be annotated to any yang-node/subpath in a given yang-module and the associated namespace(s) is retrieved based on the request URL target ,at the very beginning in translib layer before the processing of the query is done in transformer layer what happens in the following case?
> 
> Yang-module openconfig-x has following structure with list node-C annotated with name-space and the REST GET request URL is at list node-A(openconfig-x/container-As/list-A) that has no namespace associated with it and is parent of list node-C : openconfig-x container-As list-A ... ... container-Cs (child of list-A) list-C **name-space-annotation**
> 
> As per the points mentioned in this section only the default namespace will be returned since the URL is at list-A. Transformer infra processing/traversal of the child yang hierarchy happens much later in processGet(point 16.) and the node annotated with name-space will be known/reached much later. How will this be handled? The name-space seems to rely only on URL/reuest target.

The mentioned example is a bug, as we should defile namespace annotation at parent level i,e list-A

Reply:
This defies your  comment  - https://github.com/sonic-net/SONiC/pull/1701#discussion_r1812668961
that namespace annotation can be at yang subpath level. So in effect namespace has to be annotated at top level node  yang module hence its per yang module similar to pre/post-xfmrt.
Then are you saying there cannot be a use-case where list-A belongs to default/Host name space and list-C belongs to a different namespace?

Reply:
> This defies your comment - [#1701 (comment)](https://github.com/sonic-net/SONiC/pull/1701#discussion_r1812668961) that namespace annotation can be at yang subpath level. So in effect namespace has to be annotated at top level node yang module hence its per yang module similar to pre/post-xfmrt. Then are you saying there cannot be a use-case where list-A belongs to default/Host name space and list-C belongs to a different namespace?

Updated the misleading comment, yes following usecase is not supported
"Parent list-A belongs to default/Host name space and its child list-C belongs to a different namespace"